src/api/test/functional/status_controller_test.rb

```diff
@@ -54,11 +54,44 @@ def test_new_message
     assert_equal 0, messages.each.size
   end
 
+  def test_calculate_workers_by_constraints
+    post "/worker"
+    assert_response 400
+    assert_xml_tag(tag: "status", attributes: {code: "missing_parameter"})
+    post "/worker?cmd=checkconstraints&project=HiddenProject&package=TestPack&repository=10.2&arch=i586"
+    assert_response 404
+    assert_xml_tag(tag: "status", attributes: {code: "unknown_project"})
+    post "/worker?cmd=checkconstraints&project=home:Iggy&package=TestPack&repository=10.2&arch=i586"
+    assert_response :success
+    assert_select "directory" do
+      assert_select "entry", name: "x86_64:build33:1"
+    end
+    raw_post "/worker?cmd=checkconstraints&project=home:Iggy&package=TestPack&repository=10.2&arch=i586",
+             '<constraints></constraints>' # real calculation tests are done in backend test suite
+    assert_response :success
+    assert_select "directory" do
+      assert_select "entry", name: "x86_64:build33:1"
+    end
+  end
+
+  def test_worker_capability
+    get "/worker/x86_64:build33:1"
+    assert_response :success
+    assert_select "worker", hostarch: "x86_64", registerserver: "http://4.3.2.1:5253", workerid: "worker:1" do
+      assert_select "sandbox", "kvm"
+    end
+  end
+
   def test_workerstatus
-    get "/status/workerstatus"
```

Comment:
We don't document the routes anymore, but we will keep supporting them until a 3.0 release where we document the change.

I do still want to test it since this belongs to API compability to my POV.